service/client/client.js

```diff
@@ -128,8 +128,11 @@ export default class Client {
         }
 
         return {
-            headers,
-            ...options
+            ...options,
+            headers: {
```

Comment:
I see your point. Not everyone is so comfortable with lots of object destructuring in return statements.